middle_end/flambda2/types/meet_and_join_new.ml

```diff
@@ -565,6 +565,25 @@ and meet_expanded_head0 env (descr1 : ET.descr) (descr2 : ET.descr) :
 
 and meet_head_of_kind_value env (head1 : TG.head_of_kind_value)
     (head2 : TG.head_of_kind_value) : TG.head_of_kind_value meet_result =
+  let meet_is_null env (is_null1 : bool Or_unknown.t)
+      (is_null2 : bool Or_unknown.t) : _ meet_result =
+    match is_null1, is_null2 with
+    | Unknown, Unknown | Known true, Known true | Known false, Known false ->
+      Ok (Both_inputs, env)
+    | Unknown, Known _ -> Ok (Right_input, env)
+    | Known _, Unknown -> Ok (Left_input, env)
+    | Known true, Known false | Known false, Known true -> Bottom
+  in
+  combine_results2 env ~meet_a:meet_head_of_kind_value_non_null
```

Comment:
I think this is actually wrong: if both sides are nullable, but the meets in the non-null case return `Bottom`, this will return `Bottom` instead of something that is always null.

Reply:
Should be fixed in the new version